From 61c34698f9c3bd65385aa52862d6a0c6cecfbc61 Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Mon, 19 Oct 2009 10:21:50 +0800
Subject: [PATCH] x86: delete ext int unmask code

After fixing vioapic issue, ext int unmask is not
necessary all more, so delte them.

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/x86/kernel/vbi/wrhv.c |   22 ----------------------
 1 files changed, 0 insertions(+), 22 deletions(-)

diff --git a/arch/x86/kernel/vbi/wrhv.c b/arch/x86/kernel/vbi/wrhv.c
index 7ffcc9c..767f985 100644
--- a/arch/x86/kernel/vbi/wrhv.c
+++ b/arch/x86/kernel/vbi/wrhv.c
@@ -32,9 +32,6 @@
 #include <asm/cpu.h>
 #include <asm/reboot.h>		/* for struct machine_ops */
 
-#ifdef CONFIG_RTC
-#include <linux/mc146818rtc.h>
-#endif
 
 //#define WRHV_DEBUG_MSR	1
 #define WRHV_USE_XMLCONFIG	1
@@ -92,22 +89,6 @@ static void wrhv_pre_intr_init_hook(void)
 	}
 }
 
-/*by default vioapic will mask the external interrupts,
- *so umask the ext ints which you are going to use.
- */
-static inline void wrhv_umask_ext_int(void)
-{
-#ifdef CONFIG_RTC
-	/*unmask rtc irq*/
-	vbiVioapicIntUnmask(RTC_IRQ);
-#endif
-
-#ifdef CONFIG_WRHV_X86_HRTIMERS
-	vbiVioapicIntUnmask(TIMER_INT_NUM);
-#endif
-	
-}
-
 void __init wrhv_init_IRQ(void)
 {
 	int i;
@@ -149,8 +130,6 @@ void __init wrhv_init_IRQ(void)
 
 	irq_ctx_init(smp_processor_id());
 
-	wrhv_umask_ext_int();
-
 	/* race during reboot might have left a timer interrupt
 	 * pending and unacked */
 	/*
@@ -348,7 +327,6 @@ static void __devinit pci_fixup_wrhv(struct pci_dev *dev)
 		irq = vbiIntVecFind(devname, 1);
 		if (irq == VBI_INVALID_IRQ)
 			irq = WRHV_POLL_IRQ;
-		vbiIoapicIoctl(VBI_IOAPICIOCTL_UNMASK, irq, 0);
 	}
 
 	dev->irq = irq;
-- 
1.6.5.2

